src/Cool/CoolRequest.php

```diff
@@ -25,12 +25,25 @@ function getDiscovery($server) {
     }
     $disable_checks = (bool)$default_config->get('cool')['disable_cert_check'];
 
-    $stream_context = stream_context_create([
-        'ssl' => [
-            'verify_peer'       => !$disable_checks,
-            'verify_peer_name'  => !$disable_checks,
-        ]]);
-    $res = file_get_contents($discovery_url, false, $stream_context);
+    // Previously, file_get_contents() was used to fetch the discovery xml data.
+    // Depending on the environment, it can happen that file_get_contents() will
+    // hang at the end of a stream, expecting more data.
+    // With curl, this does not happen.
+    // @todo Refactor this and use e.g. Guzzle http client.
+    $curl = curl_init($discovery_url);
+    curl_setopt_array($curl, [
+        CURLOPT_RETURNTRANSFER => TRUE,
+        // Previously, when this request was done with file_get_contents() and
+        // stream_context_create(), the 'verify_peer' and 'verify_peer_name'
+        // options were set.
+        // @todo Check if an equivalent to 'verify_peer_name' exists for curl.
+        CURLOPT_SSL_VERIFYPEER => !$disable_checks,
```

Comment:
adding 
```php
        CURLOPT_SSL_VERIFYSTATUS => !$disable_checks,
        CURLOPT_SSL_VERIFYHOST => !$disable_checks,
```

seems to work.

Reply:
Can you tell me what is different with or without these additions?
I am happy to add it but I like to understand why.

Reply:
Or how we could test it.

Reply:
my setup use self signed certificates (it's local to my workstation) and without these two option it go through.

I think maybe of the two just `CURLOPT_SSL_VERIFYHOST` is needed (it doesn't work without,a nd removing `CURLOPT_SSL_VERIFYSTATUS` seems to be fine). It seems to be the equivalent of `verify_peer_name`.

Reply:
Actually it should be
```php
        CURLOPT_SSL_VERIFYHOST => $disable_checks ? 0 : 2,
```

as per https://www.php.net/manual/en/curl.constants.php#constant.curlopt-ssl-verifyhost